File: Foody/app/src/main/java/prateek_gupta/foody/ui/fragments/ingredients/IngredientsFragment.java
package prateek_gupta.foody.ui.fragments.ingredients;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;

import prateek_gupta.foody.adapters.IngredientsAdapter;
import prateek_gupta.foody.databinding.FragmentIngredientsBinding;
import prateek_gupta.foody.models.Result;
import prateek_gupta.foody.util.Constants;

public class IngredientsFragment extends Fragment {

    private final IngredientsAdapter mAdapter=new IngredientsAdapter();

    FragmentIngredientsBinding binding;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        binding= FragmentIngredientsBinding.inflate(inflater, container, false);

        if (getArguments()!=null){
            Result myBundle=getArguments().getParcelable(Constants.RECIPE_RESULT_KEY);

            setupRecyclerView();
            mAdapter.setData(myBundle.extendedIngredients);

        }

        return binding.getRoot();
    }

    void setupRecyclerView() {
        binding.ingredientsRecyclerview.setAdapter(mAdapter);
        binding.ingredientsRecyclerview.setLayoutManager(new
                LinearLayoutManager(requireContext()));
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding=null;
    }
}

File: Foody/app/src/main/java/prateek_gupta/foody/ui/fragments/ingredients/IngredientsFragment.kt
package prateek_gupta.foody.ui.fragments.ingredients

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.LinearLayoutManager
import prateek_gupta.foody.R
import prateek_gupta.foody.adapters.IngredientsAdapter
import prateek_gupta.foody.databinding.FragmentIngredientsBinding
import prateek_gupta.foody.models.Result
import prateek_gupta.foody.util.Constants.Companion.RECIPE_RESULT_KEY

class IngredientsFragment : Fragment() {

    private val mAdapter: IngredientsAdapter by lazy { IngredientsAdapter() }

    private var _binding:FragmentIngredientsBinding? = null
    private val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        // Inflate the layout for this fragment
        _binding = FragmentIngredientsBinding.inflate(inflater, container, false)

        val args = arguments
        val myBundle = args?.getParcelable(RECIPE_RESULT_KEY) as Result?

        setupRecyclerView()
        myBundle?.extendedIngredients?.let { mAdapter.setData(it) }

        return binding.root
    }

    private fun setupRecyclerView() {
        binding.ingredientsRecyclerview.adapter = mAdapter
        binding.ingredientsRecyclerview.layoutManager = LinearLayoutManager(requireContext())
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding=null
    }
}